### ISSUE TITLE
how to use a local type in tl files

### ISSUE THREAD

**Issue by virusdefender** [teal-language/tl#122]
`m.tl`

```
local event_t = record
    exename: string
end

function scan_process(): {event_t}
    
end

return {
    scan_process=scan_process
}
```

`main.tl`

```
local mtl = require("m")


function test_python_suid()
    print(mtl.scan_process()[1].exename)
end
```

`tl check main.tl` will show `unknown type event_t`

add `event_t=event_t` to module return statement does not work

change `local event_t` to `global event_t` can work

**Comment by pdesaulniers** [comment]
I believe you cannot use local types for the time being. However, you could use nested records:

m.tl:
```lua
local mtl = record
    event_t = record
        exename: string
    end
end


function mtl.scan_process(): {mtl.event_t}
    
end

return mtl
```

main.tl:
```lua
local mtl = require("m")


function test_python_suid()
    print(mtl.scan_process()[1].exename)
end
```

**Comment by virusdefender** [comment]
thank you, your code works, but if i use a different variable other than `mtl`， `tl check` will report errors, it'a a bug?

```
local mtl1 = require("m")

function test_python_suid()
    print(mtl1.scan_process()[1].exename)
end
```
```
m.tl:8:31: unknown type mtl.event_t
main.tl:5:11: cannot index something that is not a record: mtl.event_t (an unknown type)
```

**Comment by hishamhm** [comment]
@virusdefender yes, this is a buggy behavior. Note that this issue only happens becaus the function `scan_process` is empty... if you add a `return { exename = "bla" }` to the function, it type checks because processing the `return` expression causes the return type to be correctly resolved.

I'm pushing a change that will fix this (for non-generic types at least; generics need more work), and that will also change the behavior of exported values using local types.

With this change, exporting the local in the module `return` statement will work as well.

**Comment by hishamhm** [comment]
@virusdefender Thank you for the reports! It's fixed in 4105fe9.
